pkg/util/yaml/decoder.go

```diff
@@ -203,7 +206,20 @@ func (d *YAMLOrJSONDecoder) Decode(into interface{}) error {
 			d.decoder = NewYAMLToJSONDecoder(buffer)
 		}
 	}
-	return d.decoder.Decode(into)
+	err := d.decoder.Decode(into)
+	if jsonDecoder, ok := d.decoder.(*json.Decoder); ok {
```

Comment:
How does this work for yaml files? Can you add tests demonstrating that it works and reports the right lines for both json and yaml input?